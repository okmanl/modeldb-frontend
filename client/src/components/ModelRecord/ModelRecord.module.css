.model_layout {
  background-color: var(--bg-color3);
  padding: 18px 30px 30px 30px;
}

.record {
  display: grid;
  grid-template-columns: 175px auto;
  grid-template-rows: 13px auto 12px;
  border-width: 1px 0 0 0;
  border-color: var(--border-color1);
  border-style: solid;
  align-items: top;
}

.record:first-child {
  border: none;
}

.record_divider {
  grid-template-columns: auto;
}

.record_summary {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 13px auto 12px;
  align-items: top;
}

.tags {
  display: inline-block;
  padding: 5px 3px;
}

.record_summary_main {
  grid-row: 2;
  color: var(--fr-color2);
  line-height: 18px;
}

.record_name {
  font-size: 22px;
  margin-top: 5px;
}

.record_label {
  font-size: 14px;
  color: rgb(141, 141, 141);
}

.meta_block {
  padding: 5px;
  display: flex;
}

.meta_label {
  font-size: 14px;
  color: rgb(141, 141, 141);
}

.meta_value {
  padding-left: 5px;
  color: var(--fr-color2);
}

.record_summary_meta {
  grid-row: 2;
  font-size: 15px;
  line-height: 18px;
}

.record_header {
  grid-row: 2;
  color: var(--fr-color2);
  font-size: 14px;
  font-weight: 500;
  line-height: 18px;
}

.record_header_divider {
  font-size: 18px;
  margin: 5px 0;
}

.record_value {
  grid-row: 2;
  color: var(--fr-color3);
  font-size: 16px;
  line-height: 16px;
}

.record_value.name {
  font-weight: 500;
  color: var(--fr-color2);
}

.record_value.list {
  line-height: 25px;
  margin: -5px 0;
}

.tag {
  height: 34px;
  width: 100px;
  border: 1px solid var(--border-color1);
  border-radius: 2px;
  margin-right: 12px;
  float: left;
  display: grid;
  justify-items: center;
  align-items: center;
}

.tag_text {
  color: #4e4e4e;
  font-size: 13px;
  line-height: 15px;
}

.loader {
  display: flex;
  justify-content: center;
}

.record_serviceBehaviour {
}

.record_dataBehavior {
}

.record_hyperparameters {
}

.record_metrics {
}

.record_artifacts {
}

.record_datasets {
}

.record_deployInfo {
}

.notDeployedMsg {
  display: grid;
  grid-column-gap: 0.5em;
  grid-template-columns: 1em auto;
  font-size: 14px;
  font-style: italic;
  color: var(--text-color3);
  padding: 2em;
  background-color: var(--main-color1-light);
  border: solid 0.5px var(--main-color1);
  border-radius: 5px;
  white-space: pre-wrap;
  max-width: 24em;
  opacity: 0.85;
}
.notDeployedMsg_noMonitoringInfo {
}
.notDeployedMsg_noInfoInTheTimeWindow {
}
