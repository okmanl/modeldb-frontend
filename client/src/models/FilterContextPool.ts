import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { IFilterContextData } from 'store/filter';
import { IApplicationState, IThunkActionDependencies } from 'store/store';

import { IFilterData } from './Filters';
import { IMetaData } from './IMetaData';

export interface IFilterContext {
  isFilteringSupport: boolean;
  name: string;
  getMetadata(): IMetaData[];
  onSearch(
    text: string,
    dispatch: ThunkDispatch<
      IApplicationState,
      IThunkActionDependencies,
      AnyAction
    >
  ): void;
  onApplyFilters(
    filters: IFilterData[],
    dispatch: ThunkDispatch<
      IApplicationState,
      IThunkActionDependencies,
      AnyAction
    >
  ): void;
  isValidLocation(location: string): boolean;
}

class FilterContextPool {
  private contexts: { [index: string]: IFilterContext } = {};

  public registerContext(context: IFilterContext) {
    if (context.name.length === 0) {
      throw new Error('Bad context name');
    }
    if (this.contexts[context.name] !== undefined) {
      throw new Error('Context with such name is already existed');
    }

    this.contexts[context.name] = context;
  }

  public unregisterContext(name: string) {
    if (name.length === 0) {
      throw new Error('Bad context name');
    }

    if (this.contexts[name] === undefined) {
      throw new Error('Context is not registered');
    }

    delete this.contexts[name];
  }

  public initContextsData(): IFilterContextData[] {
    const result: IFilterContextData[] = [];
    for (const prop in this.contexts) {
      if (this.contexts.hasOwnProperty(prop)) {
        const ctx = this.contexts[prop];

        const resultItem = { ctx, appliedFilters: [], name: ctx.name };
        const confData = localStorage.getItem(`${ctx.name}_filter`);
        if (confData !== null) {
          resultItem.appliedFilters = JSON.parse(confData);
        }
        result.push(resultItem);
      }
    }
    return result;
  }

  public findContextByLocation(location: string): IFilterContext | undefined {
    for (const ctxName in this.contexts) {
      if (this.contexts.hasOwnProperty(ctxName)) {
        const ctx = this.contexts[ctxName];
        if (ctx.isValidLocation(location)) {
          return ctx;
        }
      }
    }
    return undefined;
  }

  public getContextByName(name: string): IFilterContext {
    if (name.length === 0) {
      throw new Error('Bad context name');
    }

    if (this.contexts[name] === undefined) {
      throw new Error('Context is not registered');
    }

    return this.contexts[name];
  }
  public hasContext(name: string) {
    return this.contexts[name] !== undefined;
  }
}

// singleton
const filterContextPool = new FilterContextPool();
export { filterContextPool as FilterContextPool };
