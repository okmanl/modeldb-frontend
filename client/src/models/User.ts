export default class User {
  public id: string | undefined;
  public name?: string | undefined;
  public email: string;
  public picture?: string | undefined;

  public constructor(id: string | undefined, email: string) {
    this.id = id;
    this.email = email;
  }

  public getNameOrEmail(): string {
    return this.name ? this.name : this.email;
  }
}

export class CurrentUser extends User {
  public dateLastLoggedIn: Date;
  public developerKey: string;

  public constructor(opts: {
    id: string | undefined;
    email: string;
    dateLastLoggedIn: Date;
    developerKey: string;
    name: string | undefined;
    picture: string | undefined;
  }) {
    const { id, email, dateLastLoggedIn, developerKey, name, picture } = opts;
    super(id, email);
    this.dateLastLoggedIn = dateLastLoggedIn;
    this.developerKey = developerKey;
    this.name = name;
    this.picture = picture;
  }
}
