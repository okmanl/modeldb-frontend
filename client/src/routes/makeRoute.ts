import { matchPath } from 'react-router';

export interface IRouteSettings<T> {
  getPath: () => string;
  getRedirectPath?: (options: T) => string;
}
export interface IRoute<T> {
  getPath: () => string;
  getRedirectPath: (options: T) => string;
  getMatch(location: string, exact?: boolean): T | null;
}

export default function makeRoute<T>({
  getPath,
  getRedirectPath,
}: IRouteSettings<T>): IRoute<T> {
  return {
    getPath,
    getRedirectPath: getRedirectPath || getPath,
    getMatch: (location: string, exact: boolean = true) => {
      const path = getPath();
      const match = matchPath<T>(location, { path, exact });
      if (match) {
        return match.params;
      }
      return null;
    },
  };
}
