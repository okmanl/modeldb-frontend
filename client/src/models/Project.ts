import { JsonObject, JsonProperty } from 'json2typescript';

import { StringToDateConverter } from 'utils/MapperConverters';

import User from './User';

@JsonObject('project')
export class Project {
  @JsonProperty('id', String)
  public id: string;
  @JsonProperty('name', String, true)
  public name: string;
  @JsonProperty('description', String, true)
  public description: string = '';
  @JsonProperty('date_created', StringToDateConverter, true)
  public dateCreated: Date = new Date();
  @JsonProperty('date_updated', StringToDateConverter, true)
  public dateUpdated: Date = new Date();
  @JsonProperty('tags', [String], true)
  public tags: string[] = [];
  @JsonProperty('owner', String, true)
  public authorId: string;
  private author: User = new User('', '');

  public constructor(id?: string, name?: string, authorId?: string) {
    this.id = id || '';
    this.name = name || '';
    this.authorId = authorId || '';
  }

  public get Author(): User {
    return this.author;
  }

  public set Author(v: User) {
    this.author = v;
  }
}
