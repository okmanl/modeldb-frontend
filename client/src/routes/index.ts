import makeRoute, { IRoute as _IRoute } from './makeRoute';

export type IRoute<T> = _IRoute<T>;

const routes = {
  mainPage: makeRoute({ getPath: () => '/' }),
  callback: makeRoute({ getPath: () => '/callback' }),
  settings: makeRoute({ getPath: () => '/settings' }),

  experimentRuns: makeRoute({
    getPath: () => '/project/:projectId/exp-runs',
    getRedirectPath: (p: { projectId: string }) =>
      `/project/${p.projectId}/exp-runs`,
  }),
  modelRecord: makeRoute({
    getPath: () => '/project/:projectId/exp-runs/:modelRecordId',
    getRedirectPath: (p: { projectId: string; modelRecordId: string }) =>
      `/project/${p.projectId}/exp-runs/${p.modelRecordId}`,
  }),
};

type ArgumentTypes<F extends Function> = F extends (...args: infer A) => any
  ? A
  : never;

export type GetRouteParams<T extends IRoute<any>> = ArgumentTypes<
  T['getRedirectPath']
>[0];
export default routes;
