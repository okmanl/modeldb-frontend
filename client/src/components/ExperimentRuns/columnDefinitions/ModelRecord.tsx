import * as React from 'react';
import { Link } from 'react-router-dom';

import routes from 'routes';

import styles from './ColumnDefs.module.css';

class ModelRecordColDef extends React.Component<any> {
  public render() {
    const { id, projectId, name } = this.props.data;
    return (
      <div className={styles.param_cell}>
        <Link
          className={styles.model_link}
          to={routes.modelRecord.getRedirectPath({
            projectId,
            modelRecordId: id,
          })}
        >
          <div className={styles.modelName_block}>
            <div className={styles.model_name}>
              {name} &nbsp;
              <i className="fa fa-external-link" />
            </div>
          </div>
        </Link>
      </div>
    );
  }
}

export default ModelRecordColDef;
