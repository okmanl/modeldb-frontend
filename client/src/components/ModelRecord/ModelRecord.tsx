import * as React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';

import Preloader from 'components/shared/Preloader/Preloader';
import tagStyles from 'components/shared/TagBlock/TagBlock.module.css';
import { IArtifact } from 'models/Artifact';
import { IHyperparameter } from 'models/HyperParameters';
import { IMetric } from 'models/Metrics';
import ModelRecord from 'models/ModelRecord';
import routes, { GetRouteParams } from 'routes';
import {
  fetchModelRecord,
  selectIsLoadingModelRecord,
  selectModelRecord,
} from 'store/model-record';
import { IApplicationState } from 'store/store';

import { bindActionCreators, Dispatch } from 'redux';
import styles from './ModelRecord.module.css';
import ShowContentBasedOnUrl from './ShowContentBasedOnUrl/ShowContentBasedOnUrl';

type IUrlProps = GetRouteParams<typeof routes.modelRecord>;

interface IPropsFromState {
  data: ModelRecord | null;
  loadingModelRecord: boolean;
}

interface IActionProps {
  fetchModelRecord: typeof fetchModelRecord;
}

type AllProps = RouteComponentProps<IUrlProps> & IPropsFromState & IActionProps;

class ModelRecordView extends React.PureComponent<AllProps> {
  public componentDidMount() {
    this.props.fetchModelRecord(this.props.match.params.modelRecordId);
  }

  public render() {
    const { data, loadingModelRecord: loading } = this.props;

    return loading ? (
      <div className={styles.loader}>
        <Preloader variant="dots" />
      </div>
    ) : data ? (
      <div className={styles.model_layout}>
        <div className={styles.record_summary}>
          <div className={styles.record_summary_main}>
            <div className={styles.record_label}>Name</div>
            <div className={styles.record_name}>{data.name}</div>
            <br />
            {data.tags && data.tags.length > 0 && (
              <div>
                <div className={styles.record_label}>Tags</div>
                <div>
                  {data.tags.map((value: string, key: number) => {
                    return (
                      <div className={styles.tags} key={key}>
                        <span className={tagStyles.staticTag}>{value}</span>
                      </div>
                    );
                  })}
                </div>
              </div>
            )}
          </div>
          <div className={styles.record_summary_meta}>
            <this.RenderModelMeta label="Id" value={data.id} />
            <this.RenderModelMeta
              label="Experiment"
              value={data.experimentId}
            />
            <this.RenderModelMeta label="Project" value={data.projectId} />
          </div>
        </div>
        {data.codeVersion! && (
          <this.Record
            header="Code version"
            additionalContainerClassName={styles.record_codeVersion}
          >
            {data.codeVersion}
          </this.Record>
        )}
        {data.hyperparameters && data.hyperparameters.length > 0 && (
          <this.Record
            header="Hyperparameters"
            additionalContainerClassName={styles.record_hyperparameters}
          >
            {data.hyperparameters.map((value: IHyperparameter, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta label={value.key} value={value.value} />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.metrics && data.metrics.length > 0 && (
          <this.Record
            header="Metrics"
            additionalContainerClassName={styles.record_metrics}
          >
            {data.metrics.map((value: IMetric, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta label={value.key} value={value.value} />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.artifacts && data.artifacts.length > 0 && (
          <this.Record
            header="Artifacts"
            additionalContainerClassName={styles.record_artifacts}
          >
            {data.artifacts.map((value: IArtifact, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta
                    label={value.key}
                    children={<ShowContentBasedOnUrl path={value.path} />}
                  />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.datasets && data.datasets.length > 0 && (
          <this.Record
            header="Datasets"
            additionalContainerClassName={styles.record_datasets}
          >
            {data.datasets.map((value: IArtifact, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta
                    label={value.key}
                    children={<ShowContentBasedOnUrl path={value.path} />}
                  />
                </div>
              );
            })}
          </this.Record>
        )}
        )}
      </div>
    ) : (
      ''
    );
  }

  // tslint:disable-next-line:function-name
  private Record(props: {
    header: string;
    children?: React.ReactNode;
    additionalValueClassName?: string;
    additionalContainerClassName?: string;
    additionalHeaderClassName?: string;
  }) {
    const {
      header,
      children,
      additionalValueClassName,
      additionalContainerClassName,
      additionalHeaderClassName,
    } = props;
    return (
      <div className={`${styles.record} ${additionalContainerClassName}`}>
        <div className={`${styles.record_header} ${additionalHeaderClassName}`}>
          {header}
        </div>
        <div className={`${styles.record_value} ${additionalValueClassName}`}>
          {children}
        </div>
      </div>
    );
  }
  // tslint:disable-next-line:function-name
  private RenderModelMeta(props: {
    label: string;
    value?: string | number;
    children?: React.ReactNode;
  }) {
    const { label, value, children } = props;
    return (
      <div className={styles.meta_block}>
        <div className={styles.meta_label}>{`${label} :`}</div>
        {value ? (
          <div className={styles.meta_value}>{value}</div>
        ) : (
          <div>{children}</div>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state: IApplicationState): IPropsFromState => {
  const modelRecord = selectModelRecord(state);
  return {
    data: modelRecord,
    loadingModelRecord: selectIsLoadingModelRecord(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): IActionProps => {
  return bindActionCreators(
    {
      fetchModelRecord,
    },
    dispatch
  );
};

export type IModelRecordProps = AllProps;
export { ModelRecordView };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModelRecordView);
