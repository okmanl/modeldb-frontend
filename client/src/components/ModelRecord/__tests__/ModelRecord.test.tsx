import { ShallowWrapper } from 'enzyme';

import { IDeployedStatusInfo, IDeployStatusInfo } from 'models/Deploy';
import ModelRecord from 'models/ModelRecord';
import routes from 'routes';
import {
  mockDataStatistics,
  mockServiceStatistics,
} from 'services/mocks/deployMock';
import { expRunsMocks } from 'services/mocks/expRunsMock';
import { makeRouterMockProps } from 'utils/tests/mocks/routeComponentProps';
import { makeShallowRenderer } from 'utils/tests/react/helpers';

import { IModelRecordProps, ModelRecordView } from '../ModelRecord';
import styles from '../ModelRecord.module.css';

const mockModelRecord: ModelRecord = {
  id: '123',
  ...expRunsMocks[0],
} as any;

const deployedStatusInfo: IDeployedStatusInfo = {
  status: 'deployed',
  data: {} as any,
};

const findRecord = (
  recordClass: string,
  component: ShallowWrapper<any, any, any>
) => {
  return component.findWhere(
    n => n.props().additionalContainerClassName === recordClass
  );
};

const makeShallowComponent = (props: Partial<IModelRecordProps> = {}) => {
  const routerMockProps = makeRouterMockProps(
    { projectId: '123', modelRecordId: mockModelRecord.id },
    {},
    {
      history: {
        location: {
          pathname: routes.modelRecord.getRedirectPath({
            projectId: '123',
            modelRecordId: mockModelRecord.id,
          }),
        },
      },
    }
  );

  const actionSpies = {
    fetchModelRecord: jest.fn(),
    checkDeployStatusUntilDeployed: jest.fn(),
    getDataStatistics: jest.fn(),
    getServiceStatistics: jest.fn(),
  };
  const defaultProps: IModelRecordProps = {
    ...routerMockProps,
    ...actionSpies,
    data: mockModelRecord,
    dataStatistics: null,
    serviceStatistics: null,
    deployState: { status: 'unknown' },
    loadingDataStatistics: false,
    loadingModelRecord: false,
    loadingServiceStatistics: false,
  };
  return {
    actionSpies,
    component: makeShallowRenderer<IModelRecordProps>(
      ModelRecordView,
      defaultProps
    )(props),
  };
};

describe('(components/ModelRecord)', () => {
  describe('on mount', () => {
    it('should load model record when there is not model record', () => {
      const { actionSpies } = makeShallowComponent({ data: null });

      expect(actionSpies.fetchModelRecord).toHaveBeenCalled();
    });

    it('should start to check deploy status until deployed when model is not deployed', () => {
      const { actionSpies: actionSpies1 } = makeShallowComponent({
        deployState: { status: 'unknown' },
      });
      expect(actionSpies1.checkDeployStatusUntilDeployed).toHaveBeenCalled();

      const { actionSpies: actionSpies2 } = makeShallowComponent({
        deployState: deployedStatusInfo,
      });
      expect(
        actionSpies2.checkDeployStatusUntilDeployed
      ).not.toHaveBeenCalled();
    });

    it('should load deploy info when model is deployed', () => {
      const { actionSpies } = makeShallowComponent({
        deployState: deployedStatusInfo,
      });

      expect(actionSpies.getDataStatistics).toHaveBeenCalled();
      expect(actionSpies.getServiceStatistics).toHaveBeenCalled();
    });
  });

  it('should start to check deploy status until deployed when model is loaded', () => {
    const { actionSpies, component } = makeShallowComponent({ data: null });

    component.setProps({ data: mockModelRecord });

    expect(actionSpies.checkDeployStatusUntilDeployed).toHaveBeenCalled();
  });

  it('should load deploy info when model is loaded, deploy status is deployed and deploy info is not loaded early', () => {
    const {
      actionSpies: actionSpies1,
      component: component1,
    } = makeShallowComponent({ data: null });

    component1.setProps({
      data: mockModelRecord,
      deployState: deployedStatusInfo,
    });

    expect(actionSpies1.getDataStatistics).toHaveBeenCalled();
    expect(actionSpies1.getServiceStatistics).toHaveBeenCalled();

    const {
      actionSpies: actionSpies2,
      component: component2,
    } = makeShallowComponent({
      data: mockModelRecord,
      deployState: { status: 'unknown' },
    });

    component2.setProps({ deployState: { status: 'deploying' } });

    expect(actionSpies2.getDataStatistics).not.toHaveBeenCalled();
    expect(actionSpies2.getServiceStatistics).not.toHaveBeenCalled();
  });

  it('should render summary', () => {
    const { component } = makeShallowComponent({ data: mockModelRecord });

    expect(component.find(`.${styles.record_summary}`)).toMatchSnapshot();
  });

  it('should render message that there are not deploy info when deploy is not deployed', () => {
    const deployStates: Array<IDeployStatusInfo | undefined> = [
      { status: 'deploying' },
      { status: 'notDeployed' },
      { status: 'unknown' },
      undefined,
    ];
    deployStates.forEach(deployState => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState,
      });
      expect(
        component.findWhere(
          wrapper => wrapper.text() === 'No monitoring information'
        )
      ).toBeTruthy();
    });
  });

  [
    { name: 'hyperparameters', className: styles.record_hyperparameters },
    { name: 'metrics', className: styles.record_metrics },
    { name: 'artifacts', className: styles.record_artifacts },
    { name: 'datasets', className: styles.record_datasets },
    { name: 'deploy info', className: styles.record_deployInfo },
  ].forEach(({ name, className }) => {
    it(`should render ${name} record`, () => {
      const { component } = makeShallowComponent({ data: mockModelRecord });

      expect(findRecord(className, component)).toMatchSnapshot();
    });
  });

  describe('when deploy status is deployed', () => {
    it('should render message that there is not deploy info in the time window when serviceStatics is undefined', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        serviceStatistics: undefined,
      });

      expect(
        component.find(`.${styles.notDeployedMsg_noInfoInTheTimeWindow}`)
      ).toMatchSnapshot();
    });

    it('should render message that there is not deploy info in the time window when serviceStatics.time is undefined', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        serviceStatistics: {
          time: undefined,
        } as any,
      });

      expect(
        component.find(`.${styles.notDeployedMsg_noInfoInTheTimeWindow}`)
      ).toMatchSnapshot();
    });

    it('should render service statistics information when service statistics is loaded', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        serviceStatistics: mockServiceStatistics,
      });

      expect(
        findRecord(styles.record_serviceBehaviour, component)
      ).toMatchSnapshot();
    });

    it('should not render service statistics information when service statistics is not loaded', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        serviceStatistics: undefined,
        loadingServiceStatistics: true,
      });

      expect(
        findRecord(styles.record_serviceBehaviour, component)
      ).toBeTruthy();
    });

    it('should render data statistics information when data statistics is loaded', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        dataStatistics: mockDataStatistics,
      });

      expect(
        findRecord(styles.record_dataBehavior, component)
      ).toMatchSnapshot();
    });

    it('should not render data statistics information when data statistics is not loaded', () => {
      const { component } = makeShallowComponent({
        data: mockModelRecord,
        deployState: deployedStatusInfo,
        dataStatistics: undefined,
        loadingDataStatistics: true,
      });

      expect(
        findRecord(styles.record_dataBehavior, component).length === 0
      ).toBeTruthy();
    });
  });
});
