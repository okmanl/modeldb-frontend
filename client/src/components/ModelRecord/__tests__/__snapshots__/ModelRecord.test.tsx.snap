// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components/ModelRecord) should render artifacts record 1`] = `
<Record
  additionalContainerClassName="record_artifacts"
  header="Artifacts"
>
  <div
    key="0"
  >
    <RenderModelMeta
      label="validation_plot"
    >
      <ShowContentBasedOnUrl
        path="../output/val_obs_11.png"
      />
    </RenderModelMeta>
  </div>
  <div
    key="1"
  >
    <RenderModelMeta
      label="model"
    >
      <ShowContentBasedOnUrl
        path="../output/tensorflow-basic_11.hdf5"
      />
    </RenderModelMeta>
  </div>
</Record>
`;

exports[`(components/ModelRecord) should render datasets record 1`] = `
<Record
  additionalContainerClassName="record_datasets"
  header="Datasets"
>
  <div
    key="0"
  >
    <RenderModelMeta
      label="train_data"
    >
      <ShowContentBasedOnUrl
        path="../data/mnist/train.npz"
      />
    </RenderModelMeta>
  </div>
</Record>
`;

exports[`(components/ModelRecord) should render deploy info record 1`] = `
<Record
  additionalContainerClassName="record_deployInfo"
  header="Deploy info"
>
  <Connect(DeployButton)
    modelId="556c26a1-781b-420f-9343-77cad7c22f9e"
  />
  <Connect(DeployManager) />
</Record>
`;

exports[`(components/ModelRecord) should render hyperparameters record 1`] = `
<Record
  additionalContainerClassName="record_hyperparameters"
  header="Hyperparameters"
>
  <div
    key="0"
  >
    <RenderModelMeta
      label="hidden_size"
      value="1024"
    />
  </div>
  <div
    key="1"
  >
    <RenderModelMeta
      label="dropout"
      value="0.4"
    />
  </div>
  <div
    key="2"
  >
    <RenderModelMeta
      label="optimizer"
      value="adam"
    />
  </div>
  <div
    key="3"
  >
    <RenderModelMeta
      label="batch_size"
      value="51257889"
    />
  </div>
  <div
    key="4"
  >
    <RenderModelMeta
      label="num_epochs"
      value="4"
    />
  </div>
  <div
    key="5"
  >
    <RenderModelMeta
      label="validation_split"
      value="0.1"
    />
  </div>
</Record>
`;

exports[`(components/ModelRecord) should render metrics record 1`] = `
<Record
  additionalContainerClassName="record_metrics"
  header="Metrics"
>
  <div
    key="0"
  >
    <RenderModelMeta
      label="train_loss"
      value="0.18684466613531112"
    />
  </div>
  <div
    key="1"
  >
    <RenderModelMeta
      label="train_acc"
      value="0.9489"
    />
  </div>
</Record>
`;

exports[`(components/ModelRecord) should render summary 1`] = `
<div
  className="record_summary"
>
  <div
    className="record_summary_main"
  >
    <div
      className="record_label"
    >
      Name
    </div>
    <div
      className="record_name"
    >
      Run 11
    </div>
    <br />
    <div>
      <div
        className="record_label"
      >
        Tags
      </div>
      <div>
        <div
          className="tags"
          key="0"
        >
          <span
            className="staticTag"
          >
            enhancement
          </span>
        </div>
        <div
          className="tags"
          key="1"
        >
          <span
            className="staticTag"
          >
            obsolete
          </span>
        </div>
        <div
          className="tags"
          key="2"
        >
          <span
            className="staticTag"
          >
            debug
          </span>
        </div>
        <div
          className="tags"
          key="3"
        >
          <span
            className="staticTag"
          >
            demo
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="record_summary_meta"
  >
    <RenderModelMeta
      label="Id"
      value="556c26a1-781b-420f-9343-77cad7c22f9e"
    />
    <RenderModelMeta
      label="Experiment"
    />
    <RenderModelMeta
      label="Project"
    />
  </div>
</div>
`;

exports[`(components/ModelRecord) when deploy status is deployed should render data statistics information when data statistics is loaded 1`] = `
<Record
  additionalContainerClassName="record_dataBehavior"
  header="Data behavior"
>
  <DeployDataChart
    height={400}
    marginBottom={40}
    marginLeft={60}
    marginRight={60}
    marginTop={60}
    modelId="556c26a1-781b-420f-9343-77cad7c22f9e"
    width={600}
  />
</Record>
`;

exports[`(components/ModelRecord) when deploy status is deployed should render message that there is not deploy info in the time window when serviceStatics is undefined 1`] = `
<div
  className="notDeployedMsg notDeployedMsg_noInfoInTheTimeWindow"
>
  <Icon
    type="exclamation-triangle"
  />
  <span>
    No monitoring information in the time window considered
  </span>
</div>
`;

exports[`(components/ModelRecord) when deploy status is deployed should render message that there is not deploy info in the time window when serviceStatics.time is undefined 1`] = `
<div
  className="notDeployedMsg notDeployedMsg_noInfoInTheTimeWindow"
>
  <Icon
    type="exclamation-triangle"
  />
  <span>
    No monitoring information in the time window considered
  </span>
</div>
`;

exports[`(components/ModelRecord) when deploy status is deployed should render service statistics information when service statistics is loaded 1`] = `
<Record
  additionalContainerClassName="record_serviceBehaviour"
  header="Service behavior"
>
  <Connect(DeployServiceChart)
    height={400}
    marginBottom={80}
    marginLeft={80}
    marginRight={80}
    marginTop={40}
    modelId="556c26a1-781b-420f-9343-77cad7c22f9e"
    width={600}
  />
</Record>
`;
